tree/treeplayer/src/TTreeProcessorMT.cxx

```diff
@@ -167,13 +168,30 @@ static ClustersAndEntries MakeClusters(const std::vector<std::string> &treeNames
       std::vector<EntryCluster> clusters;
       while ((start = clusterIter()) < entries) {
          end = clusterIter.GetNextEntry();
-         // Add the current file's offset to start and end to make them (chain) global
-         clusters.emplace_back(EntryCluster{start + offset, end + offset});
+         // Currently, if a user specified a range, the clusters will be only globally obtained
+         // Hence, the guarantee that currentStart and currentEnd are always as expected
+         // That are: the cluster start must be at least startEntry, and at most the endEntry
```

Comment:
sorry, I can't parse this. what do you mean here? in particular I don't know what it means that `currentStart` and `currentEnd` "are always as expected", and I don't understand whether we have to _enforce_ the conditions in the second line or we have a _guarantee_ that the conditions are satisfied